Tests/CSharp/TypeCastTests.cs

```diff
@@ -1313,11 +1313,11 @@ internal partial class CastTest
     }
     private decimal? Test2(object input)
     {
-        return (decimal?)input;
+        return (decimal?)(double?)input;
```

Comment:
In this case I think you've fixed a bug. previously a self verifying test would have shown up this subtle bug if it passed in a very long precise decimal to the method, and asserted the slightly different value coming out from the VB version.